sunpy/instr/lyra.py

```diff
@@ -1,123 +1,625 @@
+        if channels:
+            for i, f in enumerate(clean_channels):
+                clean_channels[i] = np.delete(f, bad_indices)
+    # If return_artifacts kwarg is True, return a list containing
+    # information on what artifacts found, removed, etc.  See docstring.
+    if return_artifacts:
+        artifact_status = {"lytaf": lytaf,
+                           "removed": lytaf[artifact_indices],
+                           "not_removed": np.delete(lytaf, artifact_indices),
+                           "not_found": artifacts_not_found}
+    # Output FITS file if fits kwarg is set
+    if fitsfile:
+        # Create time array of time strings rather than datetime objects
+        # and verify filecolumns have been correctly input.  If None,
+        # generate generic filecolumns (see docstring of function called
+        # below.
+        string_time, filecolumns = _prep_columns(time, channels, filecolumns)
+        # Prepare column objects.
+        cols = [fits.Column(name=filecolumns[0], format="26A",
+                            array=string_time)]
+        if channels:
+            for i, f in enumerate(channels):
+                cols.append(fits.Column(name=filecolumns[i+1], format="D",
+                                        array=f))
+        coldefs = fits.ColDefs(cols)
+        tbhdu = fits.new_table(coldefs)
+        hdu = fits.PrimaryHDU()
+        tbhdulist = fits.HDUList([hdu, tbhdu])
+        # Write data to fits file.
+        tbhdulist.writeto(fitsfile)
+    # Output csv file if fits kwarg is set.
+    if csvfile:
+        # Create time array of time strings rather than datetime objects
+        # and verify filecolumns have been correctly input.  If None,
+        # generate generic filecolumns (see docstring of function called
+        # below.
+        string_time, filecolumns = _prep_columns(time, channels, filecolumns)
+        # Open and write data to csv file.
+        with open(csvfile, 'w') as openfile:
+            csvwriter = csv.writer(openfile, delimiter=';')
+            # Write header.
+            csvwriter.writerow(filecolumns)
+            # Write data.
+            if not channels:
+                for i in range(len(time)):
+                    csvwriter.writerow(string_time[i])
+            else:
+                for i in range(len(time)):
+                    row = [string_time[i]]
+                    for f in channels:
+                        row.append(f[i])
+                    csvwriter.writerow(row)
+    # Return values.
+    if return_artifacts:
+        if not channels:
+            return clean_time, artifact_status
+        else:
+            return clean_time, clean_channels, artifact_status
+    else:
+        if not channels:
+            return clean_time
+        else:
+            return clean_time, clean_channels
+
+def get_lytaf_events(start_time, end_time, lytaf_path=None,
+                     combine_files=["lyra", "manual", "ppt", "science"],
```

Comment:
I would make this combine_files a tuple here to protect it from trouble later on.